### DISCUSSION TITLE
Data sent to Livewire component via alpine being wiped out when the last value has an empty string

### DISCUSSION THREAD

**Discussion by VitorTS** [livewire/livewire#7872] [7. Bugs (archived)]
### Livewire version

3.4.4

### Laravel version

10.43.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1 - make an alpine method that calls a livewire method via `this.$wire`
2 - send data to livewire via `this.$wire.set('<name>',<value>, true)` 
3 - make sure the last value has an empty string

### Repository or Wirebox URL

https://wirebox.app/b/4lo6g

### Additional information

This bug happens because of a bug inside in `mergeQueuedUpdates(diff)` in the file `js/component.js`:

```js
    mergeQueuedUpdates(diff) {
        Object.entries(this.queuedUpdates).forEach(([updateKey, updateValue]) => {
            Object.entries(diff).forEach(([diffKey, diffValue]) => {
                if (diffKey.startsWith(updateValue)) { // the bug happens here
                    delete diff[diffKey]
                }
            })

            diff[updateKey] = updateValue
        })

        this.queuedUpdates = []

        return diff
    }
```

You can reproduce this bug like this:

```js
mergeQueuedUpdates.call({
    queuedUpdates: {
      value1: 'A',
      value2: 'B',
      value3: '',
    }
},
{
  value1: 'A',
  value2: 'B',
  value3: '',
})
```

it will return `{value3: ''}`

for the last parameter (value3) `updateValue` is an empty string, so for every key in the object (value1, value2, value3) we are going to run `diffKey.startsWith("")` which *always* returns true. Leading to the `diff` object being deleted.

The solution is to change `if (diffKey.startsWith(updateValue))` to `if (updateValue !== '' && diffKey.startsWith(updateValue))`.

I would make a PR for it myself but I can't get the tests to run despite following your guide. So hopefully somebody else can make a PR for this.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@VitorTS any chance you could submit a failing test PR? Thanks!

**Reply by timlw** [reply]
@joshhanley any chance we can implement the suggested fix from @VitorTS ? Can confirm this resolves the issue.

**Reply by joshhanley** [reply]
@timlw yep all PRs are welcome! 🙂
